### PR TITLE
[BISERVER-11584] - Data source created via Data Source Wizard on one …

### PR DESCRIPTION
**Pull request by andrei-viaryshka**
…cluster node is not visible on the other nodes

Add option to disable domainIds caching.

### CODE REVIEW

**Comment by andrei-viaryshka** [comment]
@pentaho-nbaker @mdamour1976 Could you please review?

File: extensions/src/org/pentaho/platform/plugin/services/metadata/SessionCachingMetadataDomainRepository.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
  * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  * See the GNU Lesser General Public License for more details.
  *
- * Copyright (c) 2002-2013 Pentaho Corporation..  All rights reserved.
+ * Copyright (c) 2002-2016 Pentaho Corporation..  All rights reserved.
```
Comment (dkincade) [inline extensions/src/org/pentaho/platform/plugin/services/metadata/SessionCachingMetadataDomainRepository.java]: This is a sample comment attached to the date change - please ignore

**Comment by andrei-viaryshka** [comment]
@pentaho-nbaker @mdamour1976 @pamval Could you please review?

**Comment by brosander** [comment]
http://wingman.pentaho.com:8080/job/wingman/19/

**Comment by brosander** [comment]
Error building command: Unable to determine common build directory for assembly/package-res/biserver/pentaho-solutions/system/system.properties

**Comment by lucboudreau** [comment]
http://10.100.11.222:8080/job/wingman-ant-cmd/9/

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. ant -f extensions/build.xml -Dtestreports.dir=bin/reports/unit-test -Djunit.jvmargs=-XX:PermSize=256m clean-all resolve jacoco && ant -f extensions/build.xml -Dpentaho.coding.standards.version=1.0.4 -Dtestreports.dir=bin/reports/integration-test -Djunit.jvmargs=-XX:PermSize=256m jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove



Changed files:
==============
  assembly/package-res/biserver/pentaho-solutions/system/system.properties
  extensions/src/org/pentaho/platform/plugin/services/metadata/SessionCachingMetadataDomainRepository.java
  extensions/test-src/org/pentaho/platform/plugin/services/metadata/SessionCachingMetadataDomainRepositoryTest.java

Still Broken Tests:
======================
org.pentaho.platform.osgi.SystemPackageExtrapolatorTest.testJBossProvider:
-----------------------------------------------
junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError: 
Expected: iterable containing ["org.apache", "org.apache.log4j"]
     but: item 0: was "org.apache.log4j"
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
	at org.pentaho.platform.osgi.SystemPackageExtrapolatorTest.testJBossProvider(SystemPackageExtrapolatorTest.java:85)

```
org.pentaho.platform.plugin.services.importer.ArchiveLoaderTest.testWillContinueToLoadOnException:
-----------------------------------------------
org.mockito.exceptions.verification.junit.ArgumentsAreDifferent
```
Argument(s) are different! Wanted:
file.renameTo(
    
);
-> at org.pentaho.platform.plugin.services.importer.ArchiveLoaderTest.testWillContinueToLoadOnException(ArchiveLoaderTest.java:93)
Actual invocation has different arguments:
file.renameTo(
    /root/path/jobs.zip.197001021017
);
-> at org.pentaho.platform.plugin.services.importer.ArchiveLoader.loadAll(ArchiveLoader.java:68)

	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.pentaho.platform.plugin.services.import
```
org.pentaho.platform.plugin.services.importer.ArchiveLoaderTest.testWillImportAllZipsInADirectory:
-----------------------------------------------
org.mockito.exceptions.verification.junit.ArgumentsAreDifferent
```
Argument(s) are different! Wanted:
file.renameTo(
    
);
-> at org.pentaho.platform.plugin.services.importer.ArchiveLoaderTest.testWillImportAllZipsInADirectory(ArchiveLoaderTest.java:66)
Actual invocation has different arguments:
file.renameTo(
    /root/path/jobs.zip.197001021017
);
-> at org.pentaho.platform.plugin.services.importer.ArchiveLoader.loadAll(ArchiveLoader.java:68)

	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.pentaho.platform.plugin.services.import
```

Unit test coverage change:
============
org.pentaho.platform.config.MondrianConfigProperties:
 * Branch Change :  + 7.14%
 * Instruction Change :  + 3.16%

org.pentaho.platform.plugin.services.metadata.PentahoMetadataDomainRepository:
 * Branch Change : **-1.25%**
 * Complexity Change : **-.80%**
 * Instruction Change : **-.93%**
 * Line Change : **-.82%**

org.pentaho.platform.plugin.services.metadata.SessionCachingMetadataDomainRepository:
 * Branch Change :  + 7.97%
 * Complexity Change :  + 4.50%
 * Instruction Change :  + 10.07%
 * Line Change :  + 10.68%
 * Method Change :  + 6.90%



Integration test coverage change:
============
org.pentaho.platform.plugin.services.metadata.SessionCachingMetadataDomainRepository:
 * Branch Change : **-1.43%**
 * Complexity Change : **-2.67%**
 * Instruction Change : **-.92%**
 * Line Change : **-.22%**

**Comment by eliofreitas** [comment]
Rejected from automatic build ,authorized users only

**Pull request merged**